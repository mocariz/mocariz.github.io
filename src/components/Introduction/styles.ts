import styled from 'styled-components'

export const Wrapper = styled.main`
  color: #fff;
  width: 100%;
  height: 100%;
  padding-top: 3rem;
  text-align: center;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
`

export const Block = styled.div`
  width: 50%;
`
export const Description = styled.h2`
  font-style: normal;
  font-weight: 500;
  font-size: 32px;
  line-height: 37px;
  color: #bb6bd9;

  b {
    font-weight: 700;
  }
`

export const BlockLinks = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  width: 250px;
  margin: auto;
  padding-top: 36px;
`

export const Link = styled.a`
  color: #11b488;
  cursor: pointer;
`

export const ArrowheadDown = styled.div`
  margin-top: 150px;
  cursor: pointer;
`
