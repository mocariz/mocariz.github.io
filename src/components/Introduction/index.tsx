import * as S from './styles'
import { Github, Linkedin } from '@styled-icons/boxicons-logos'
import { Mail } from '@styled-icons/entypo'

const Main = () => (
  <S.Wrapper>
    <S.Block>
      <S.Description>
        Olá, meu nome é <b>Mônica</b> e eu sou <b>Desenvolvedora Front-end</b>.
      </S.Description>
      <S.BlockLinks>
        <S.Link href="https://github.com/mocariz" target="_blank">
          <Github size={52} aria-label="Github" />
        </S.Link>
        <S.Link href="https://www.linkedin.com/in/mocariz/" target="_blank">
          <Linkedin size={52} aria-label="Linkedin" />
        </S.Link>
        <S.Link href="mailto:mota.ocariz@gmail.com">
          <Mail size={52} aria-label="Mail" />
        </S.Link>
      </S.BlockLinks>
    </S.Block>
    <S.Block>
      <S.Illustration
        src="/img/banner.svg"
        alt="A developer sitting at the table with a notebook on her lap, beside her is her little dog Tessa, a white poodle."
      />
    </S.Block>
  </S.Wrapper>
)

export default Main
