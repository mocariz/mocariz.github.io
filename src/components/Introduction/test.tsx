import { render, screen } from '@testing-library/react'

import Introduction from '.'

describe('<Introduction />', () => {
  it('should render the heading', () => {
    const { container } = render(<Introduction />)

    expect(
      screen.getByRole('heading', { name: /Olá, meu nome é Mônica e eu sou Desenvolvedora Front-end./i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
