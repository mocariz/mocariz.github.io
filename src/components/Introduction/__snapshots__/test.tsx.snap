// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Introduction /> should render the heading 1`] = `
.c0 {
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  width: 50%;
}

.c2 {
  font-style: normal;
  font-weight: 500;
  font-size: 32px;
  line-height: 37px;
  color: #BB6BD9;
}

.c2 b {
  font-weight: 700;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 250px;
  margin: auto;
  padding-top: 36px;
}

.c4 {
  color: #11B488;
  cursor: pointer;
}

.c6 {
  margin-top: 3rem;
}

.c5 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

<main
  class="c0"
>
  <div
    class="c1"
  >
    <h2
      class="c2"
    >
      Olá, meu nome é 
      <b>
        Mônica
      </b>
       e eu sou 
      <b>
        Desenvolvedora Front-end
      </b>
      .
    </h2>
    <div
      class="c3"
    >
      <a
        class="c4"
        href="https://github.com/mocariz"
        target="_blank"
      >
        <svg
          aria-hidden="true"
          aria-label="Github"
          class="c5"
          fill="currentColor"
          focusable="false"
          height="52"
          viewBox="0 0 24 24"
          width="52"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M12.026 2c-5.509 0-9.974 4.465-9.974 9.974 0 4.406 2.857 8.145 6.821 9.465.499.09.679-.217.679-.481 0-.237-.008-.865-.011-1.696-2.775.602-3.361-1.338-3.361-1.338-.452-1.152-1.107-1.459-1.107-1.459-.905-.619.069-.605.069-.605 1.002.07 1.527 1.028 1.527 1.028.89 1.524 2.336 1.084 2.902.829.091-.645.351-1.085.635-1.334-2.214-.251-4.542-1.107-4.542-4.93 0-1.087.389-1.979 1.024-2.675-.101-.253-.446-1.268.099-2.64 0 0 .837-.269 2.742 1.021a9.582 9.582 0 0 1 2.496-.336 9.554 9.554 0 0 1 2.496.336c1.906-1.291 2.742-1.021 2.742-1.021.545 1.372.203 2.387.099 2.64.64.696 1.024 1.587 1.024 2.675 0 3.833-2.33 4.675-4.552 4.922.355.308.675.916.675 1.846 0 1.334-.012 2.41-.012 2.737 0 .267.178.577.687.479C19.146 20.115 22 16.379 22 11.974 22 6.465 17.535 2 12.026 2z"
            fill-rule="evenodd"
          />
        </svg>
      </a>
      <a
        class="c4"
        href="https://www.linkedin.com/in/mocariz/"
        target="_blank"
      >
        <svg
          aria-hidden="true"
          aria-label="Linkedin"
          class="c5"
          fill="currentColor"
          focusable="false"
          height="52"
          viewBox="0 0 24 24"
          width="52"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="4.983"
            cy="5.009"
            r="2.188"
          />
          <path
            d="M9.237 8.855v12.139h3.769v-6.003c0-1.584.298-3.118 2.262-3.118 1.937 0 1.961 1.811 1.961 3.218v5.904H21v-6.657c0-3.27-.704-5.783-4.526-5.783-1.835 0-3.065 1.007-3.568 1.96h-.051v-1.66H9.237zm-6.142 0H6.87v12.139H3.095z"
          />
        </svg>
      </a>
      <a
        class="c4"
        href="mailto:mota.ocariz@gmail.com"
      >
        <svg
          aria-hidden="true"
          aria-label="Mail"
          class="c5"
          fill="currentColor"
          focusable="false"
          height="52"
          viewBox="0 0 20 20"
          width="52"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1.574 5.286l7.5 4.029c.252.135.578.199.906.199.328 0 .654-.064.906-.199l7.5-4.029c.489-.263.951-1.286.054-1.286H1.521c-.897 0-.435 1.023.053 1.286zm17.039 2.203l-7.727 4.027c-.34.178-.578.199-.906.199s-.566-.021-.906-.199-7.133-3.739-7.688-4.028C.996 7.284 1 7.523 1 7.707V15c0 .42.566 1 1 1h16c.434 0 1-.58 1-1V7.708c0-.184.004-.423-.387-.219z"
          />
        </svg>
      </a>
    </div>
  </div>
  <div
    class="c1"
  >
    <img
      alt="A developer sitting at the table with a notebook on her lap, beside her is her little dog Tessa, a white poodle."
      class="c6"
      src="/img/banner.svg"
    />
  </div>
</main>
`;
